src/modules/0.0.22/aws_ec2/mappers/general_purpose_volume.ts

```diff
@@ -38,14 +38,9 @@ export class GeneralPurposeVolumeMapper extends MapperBase<GeneralPurposeVolume>
     if (!vol?.VolumeId) return undefined;
     out.volumeId = vol.VolumeId;
     out.volumeType = vol.VolumeType as GeneralPurposeVolumeType;
-    // todo: find by region too when multiregion
     out.availabilityZone =
-      (await awsVpcModule.availabilityZone.db.read(ctx)).find(
-        (az: AvailabilityZone) => az.name === vol.AvailabilityZone,
-      ) ??
-      (await awsVpcModule.availabilityZone.cloud.read(ctx)).find(
-        (az: AvailabilityZone) => az.name === vol.AvailabilityZone,
-      );
+      (await awsVpcModule.availabilityZone.db.read(ctx, super.generateId(vol.AvailabilityZone, region))) ??
```

Comment:
You shouldn't need to explicitly call `super.generateId`, right? It should still work with `this.generateId`.

Reply:
I have to test it but yes it should work, I've pushed to see if everything keeps working as expected